db/db_impl/db_impl_write.cc

```diff
@@ -1933,11 +2107,44 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
   }
 
   cfd->mem()->SetNextLogNumber(logfile_number_);
-  cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
+  // If MemPurge activated, purge and delete current memtable.
+  if (immutable_db_options_.experimental_allow_mempurge &&
```

Comment:
I think this PR doesn't handle the WAL full case right?  When WAL is full, we may want to skip the purge and flush the memtable anyway.

Reply:
Correct, I left all WAL handlings for later (Milestone 1C, "Getting MemPurge to work with WAL"). However following your comment I just added a check on the FlushReason so that if the flush operation happens because of a WAL full or else, the MemPurge operation is not triggered and a regular flush operation will take place instead.